src/BigDecimal.php

```diff
@@ -202,6 +210,14 @@ public function multipliedBy($that) : BigDecimal
             return $this;
         }
 
+        if ($this->value === '1' && $this->scale === 0) {
+            return $that;
+        }
+
+        if ($that->value === '0' || $this->value === '0') {
+            return new BigDecimal('0', $this->scale + $that->scale);
+        }
```

Comment:
Not sure about this one: the optimizations above are really to return one of the objects without modification whenever possible, but for the other micro-optimizations, they're [already handled by the calculator](https://github.com/brick/math/blob/master/src/Internal/Calculator/NativeCalculator.php#L102-L104) if needed.

I don't think this extra check brings much value.